### ISSUE TITLE
Epic 3: Forecasting Model Development

### ISSUE THREAD

**Issue by Hlumelo-Ncedana** [Hlumelo-Ncedana/Supply-Chain-Optimization-via-Demand-Forecasting#10]
3.3: As a data scientist, I want to develop machine learning models to leverage all engineered features
Tasks:

Create proper time-series train/validation split

Implement Random Forest regressor

Train XGBoost model with hyperparameter tuning

Implement LightGBM for faster training

Evaluate all ML models using WMAE metric
